### ISSUE TITLE
Slack::RealTime::Client::ClientNotStartedError with latest master and 

### ISSUE THREAD

**Issue by rmoriz** [slack-ruby/slack-ruby-client#233]
(may be related to #231)

using the current master branch version (ref 8702c27c749f), our bot is crashing almost every 30 minutes with the following exception when using `async-websocket` (we just switched from `celluloid-io`):

```
I, [2018-10-27T03:38:57.558971 #6]  INFO -- : Successfully connected team Team xxx (xxx) to https://xxx.slack.com.
I, [2018-10-27T03:38:57.559302 #6]  INFO -- : Health Check started
#<Thread:0x000056103ec86eb0@/usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/concurrency/async.rb:22 run> terminated with exception (report_on_exception is true):
/usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/client.rb:108:in `run_ping!': Slack::RealTime::Client::ClientNotStartedError (Slack::RealTime::Client::ClientNotStartedError)
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/concurrency/async.rb:23:in `block (2 levels) in start_async'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/timer.rb:98:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:43:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:81:in `block in fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `reverse_each'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:95:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:80:in `wait'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/async-1.10.3/lib/async/reactor.rb:152:in `run'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/concurrency/async.rb:24:in `block in start_async'
/usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/client.rb:108:in `run_ping!': Slack::RealTime::Client::ClientNotStartedError (Slack::RealTime::Client::ClientNotStartedError)
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/concurrency/async.rb:23:in `block (2 levels) in start_async'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/timer.rb:98:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:43:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:81:in `block in fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `reverse_each'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:95:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:80:in `wait'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/async-1.10.3/lib/async/reactor.rb:152:in `run'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-8702c27c749f/lib/slack/real_time/concurrency/async.rb:24:in `block in start_async'
```

**Comment by dblock** [comment]
I think this is the same as discussed in https://github.com/slack-ruby/slack-ruby-client/issues/208#issuecomment-433309979

@RodneyU215 you should PR your `remove_retry_ping_logic` branch

@rmoriz Give `gem 'slack-ruby-client', github: 'RodneyU215/slack-ruby-client', branch: 'remove_retry_ping_logic'` a try?

**Comment by rmoriz** [comment]
Now we are just getting reconnects every couple of minutes… :(

```
I, [2018-10-27T18:11:11.006739 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
I, [2018-10-27T18:20:12.343877 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
I, [2018-10-27T18:22:13.374668 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
I, [2018-10-27T18:27:27.648527 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
I, [2018-10-27T18:40:29.014639 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
I, [2018-10-27T18:46:30.230051 #6]  INFO -- : Successfully connected team Team xxx (yyy) to https://xxxkit.slack.com.
```

**Comment by dblock** [comment]
@rmoriz This could use some debugging ... care to set logger level to DEBUG and see what this loop looks like?

**Comment by dblock** [comment]
This is closed via #234. The problem above in the comment possibly not - let us know @rmoriz.

**Comment by rmoriz** [comment]
Here is a debug log:

```
D, [2018-11-01T16:10:19.537403 #7] DEBUG -- Slack::RealTime::Concurrency::Async::Socket#send_data: {"type":"ping","id":8}
D, [2018-11-01T16:10:19.543877 #7] DEBUG -- SlackRubyBot::Client#run_loop: WebSocket::Driver::MessageEvent, {"type":"pong","reply_to":8}
D, [2018-11-01T16:10:19.543953 #7] DEBUG -- SlackRubyBot::Client#dispatch: reply_to=8, type=pong
D, [2018-11-01T16:11:19.561331 #7] DEBUG -- : SlackRubyBot::Client#restart_async
I, [2018-11-01T16:11:19.562078 #7]  INFO -- request: POST https://slack.com/api/rtm.start
D, [2018-11-01T16:11:19.562122 #7] DEBUG -- request: Accept: "application/json; charset=utf-8"
User-Agent: "Slack Ruby Client/0.13.2"
Content-Type: "application/x-www-form-urlencoded"
#<Thread:0x000055766efaf6d8@/usr/src/app/catbot/services/health_check.rb:17 run> terminated with exception (report_on_exception is true):
/usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-a883e4671d3a/lib/slack/real_time/client.rb:164:in `send_json': Slack::RealTime::Client::ClientNotStartedError (Slack::RealTime::Client::ClientNotStartedError)
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-a883e4671d3a/lib/slack/real_time/api/ping.rb:11:in `ping'
	from /usr/src/app/catbot/services/health_check.rb:24:in `block (2 levels) in call'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/timer.rb:98:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:43:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:81:in `block in fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `reverse_each'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:95:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:80:in `wait'
	from /usr/src/app/catbot/services/health_check.rb:26:in `block (2 levels) in call'
	from /usr/src/app/catbot/services/health_check.rb:26:in `loop'
	from /usr/src/app/catbot/services/health_check.rb:26:in `block in call'
/usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-a883e4671d3a/lib/slack/real_time/client.rb:164:in `send_json': Slack::RealTime::Client::ClientNotStartedError (Slack::RealTime::Client::ClientNotStartedError)
	from /usr/src/app/vendor/bundle/ruby/2.5.0/bundler/gems/slack-ruby-client-a883e4671d3a/lib/slack/real_time/api/ping.rb:11:in `ping'
	from /usr/src/app/catbot/services/health_check.rb:24:in `block (2 levels) in call'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/timer.rb:98:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:43:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:81:in `block in fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `reverse_each'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/events.rb:80:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:95:in `fire'
	from /usr/src/app/vendor/bundle/ruby/2.5.0/gems/timers-4.1.2/lib/timers/group.rb:80:in `wait'
	from /usr/src/app/catbot/services/health_check.rb:26:in `block (2 levels) in call'
	from /usr/src/app/catbot/services/health_check.rb:26:in `loop'
	from /usr/src/app/catbot/services/health_check.rb:26:in `block in call'
I, [2018-11-01T16:11:21.645985 #7]  INFO -- request: POST https://slack.com/api/rtm.start
D, [2018-11-01T16:11:21.646040 #7] DEBUG -- request: Accept: "application/json; charset=utf-8"
User-Agent: "Slack Ruby Client/0.13.2"
Content-Type: "application/x-www-form-urlencoded"
I, [2018-11-01T16:11:22.151282 #7]  INFO -- response: Status 200
D, [2018-11-01T16:11:22.151360 #7] DEBUG -- response: content-type: "application/json; charset=utf-8"
content-length: "37792"
connection: "close"
date: "Thu, 01 Nov 2018 16:11:21 GMT"
server: "Apache"
x-accepted-oauth-scopes: "rtm:stream,client"
x-content-type-options: "nosniff"
x-slack-router: "p"
expires: "Mon, 26 Jul 1997 05:00:00 GMT"
cache-control: "private, no-cache, no-store, must-revalidate"
x-oauth-scopes: "identify,read,post,client,apps"
pragma: "no-cache"
x-xss-protection: "0"
x-slack-req-id: "ab2accf1-632e-419d-b47b-6b794e18a0f8"
x-slack-exp: "1"
x-slack-backend: "h"
referrer-policy: "no-referrer"
strict-transport-security: "max-age=31536000; includeSubDomains; preload"
vary: "Accept-Encoding"
access-control-allow-origin: "*"
x-via: "haproxy-www-50ho"
x-cache: "Miss from cloudfront"
via: "1.1 c55f09a9188f77960d35c97bad15e1b2.cloudfront.net (CloudFront)"
x-amz-cf-id: "OweNR3qqJnCEXg-02eDnHsfJy2DRROdzWoKB6uBtMIU__nh_cGX71Q=="
D, [2018-11-01T16:11:22.268157 #7] DEBUG -- Slack::RealTime::Concurrency::Async::Socket#connect!: Slack::RealTime::Concurrency::Async::Client
D, [2018-11-01T16:11:22.276244 #7] DEBUG -- SlackRubyBot::Client#run_loop: WebSocket::Driver::OpenEvent
D, [2018-11-01T16:11:22.452187 #7] DEBUG -- SlackRubyBot::Client#run_loop: WebSocket::Driver::MessageEvent, {"type": "hello"}
D, [2018-11-01T16:11:22.452281 #7] DEBUG -- SlackRubyBot::Client#dispatch: type=hello
I, [2018-11-01T16:11:22.452322 #7]  INFO -- : Successfully connected team Team YYYYY (XXXXXX) to https://XXXXX.slack.com.
I, [2018-11-01T16:11:22.452448 #7]  INFO -- : Health Check started
```

`catbot/services/health_check.rb` does the following:

```ruby
require 'timers'

module Catbot
  module Services
    class SlackConnectionError < RuntimeError; end

    class HealthCheck
      attr_reader :client

      def call(client, _data)
        @client = client

        # not sure if this is overkill, celluloid actor in the main thread should
        # keep slack open...
        Thread.new do
          $stdout.sync = true
          @logger = Logger.new($stdout)
          @logger.info 'Health Check started'

          timers = Timers::Group.new
          timers.now_and_every(120) do
            raise SlackConnectionError, 'slack connection failed' unless client.ping
          end
          loop { timers.wait }
        end
      end
    end
  end
end
```
